### ISSUE TITLE
Completion of method definition

### ISSUE THREAD

**Issue by ColinDuquesnoy** [davidhalter/jedi#585]
When writing a method definition, jedi should provide the list of parent method names instead of the list of namespace's definitions.

```python
Python 3.4.3 (default, Mar 26 2015, 22:03:40) 
[GCC 4.9.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import jedi
>>> code = 'class Foo:\n    def '
>>> print(jedi.Script(code).completions())
[<Completion: abs>, <Completion: all>, <Completion: and>, <Completion: any>, <Completion: ArithmeticError>, <Completion: as>, <Completion: ascii>, <Completion: assert>, <Completion: AssertionError>, <Completion: AttributeError>, <Completion: BaseException>, <Completion: bin>, <Completion: BlockingIOError>, <Completion: bool>, <Completion: break>, <Completion: BrokenPipeError>, <Completion: BufferError>, <Completion: bytearray>, <Completion: bytes>, <Completion: BytesWarning>, <Completion: callable>, <Completion: ChildProcessError>, <Completion: chr>, <Completion: class>, <Completion: classmethod>, <Completion: compile>, <Completion: complex>, <Completion: ConnectionAbortedError>, <Completion: ConnectionError>, <Completion: ConnectionRefusedError>, <Completion: ConnectionResetError>, <Completion: continue>, <Completion: copyright>, <Completion: credits>, <Completion: def>, <Completion: del>, <Completion: delattr>, <Completion: DeprecationWarning>, <Completion: dict>, <Completion: dir>, <Completion: divmod>, <Completion: elif>, <Completion: Ellipsis>, <Completion: else>, <Completion: enumerate>, <Completion: EnvironmentError>, <Completion: EOFError>, <Completion: eval>, <Completion: except>, <Completion: Exception>, <Completion: exec>, <Completion: exit>, <Completion: False>, <Completion: FileExistsError>, <Completion: FileNotFoundError>, <Completion: filter>, <Completion: finally>, <Completion: float>, <Completion: FloatingPointError>, <Completion: Foo>, <Completion: for>, <Completion: format>, <Completion: from>, <Completion: frozenset>, <Completion: FutureWarning>, <Completion: GeneratorExit>, <Completion: getattr>, <Completion: global>, <Completion: globals>, <Completion: hasattr>, <Completion: hash>, <Completion: help>, <Completion: hex>, <Completion: id>, <Completion: if>, <Completion: import>, <Completion: ImportError>, <Completion: ImportWarning>, <Completion: in>, <Completion: IndentationError>, <Completion: IndexError>, <Completion: input>, <Completion: int>, <Completion: InterruptedError>, <Completion: IOError>, <Completion: is>, <Completion: IsADirectoryError>, <Completion: isinstance>, <Completion: issubclass>, <Completion: iter>, <Completion: KeyboardInterrupt>, <Completion: KeyError>, <Completion: lambda>, <Completion: len>, <Completion: license>, <Completion: list>, <Completion: locals>, <Completion: LookupError>, <Completion: map>, <Completion: max>, <Completion: MemoryError>, <Completion: memoryview>, <Completion: min>, <Completion: NameError>, <Completion: next>, <Completion: None>, <Completion: nonlocal>, <Completion: not>, <Completion: NotADirectoryError>, <Completion: NotImplemented>, <Completion: NotImplementedError>, <Completion: object>, <Completion: oct>, <Completion: open>, <Completion: or>, <Completion: ord>, <Completion: OSError>, <Completion: OverflowError>, <Completion: pass>, <Completion: PendingDeprecationWarning>, <Completion: PermissionError>, <Completion: pow>, <Completion: print>, <Completion: ProcessLookupError>, <Completion: property>, <Completion: quit>, <Completion: raise>, <Completion: range>, <Completion: ReferenceError>, <Completion: repr>, <Completion: ResourceWarning>, <Completion: return>, <Completion: reversed>, <Completion: round>, <Completion: RuntimeError>, <Completion: RuntimeWarning>, <Completion: set>, <Completion: setattr>, <Completion: slice>, <Completion: sorted>, <Completion: staticmethod>, <Completion: StopIteration>, <Completion: str>, <Completion: sum>, <Completion: super>, <Completion: SyntaxError>, <Completion: SyntaxWarning>, <Completion: SystemError>, <Completion: SystemExit>, <Completion: TabError>, <Completion: TimeoutError>, <Completion: True>, <Completion: try>, <Completion: tuple>, <Completion: type>, <Completion: TypeError>, <Completion: UnboundLocalError>, <Completion: UnicodeDecodeError>, <Completion: UnicodeEncodeError>, <Completion: UnicodeError>, <Completion: UnicodeTranslateError>, <Completion: UnicodeWarning>, <Completion: UserWarning>, <Completion: ValueError>, <Completion: vars>, <Completion: Warning>, <Completion: while>, <Completion: with>, <Completion: yield>, <Completion: ZeroDivisionError>, <Completion: zip>, <Completion: __build_class__>, <Completion: __debug__>, <Completion: __doc__>, <Completion: __file__>, <Completion: __import__>, <Completion: __loader__>, <Completion: __name__>, <Completion: __package__>, <Completion: __spec__>]
>>> 
```

E.g., here we should get ```['__abs__',  '__add__',  '__eq__', ...]```

**Comment by davidhalter** [comment]
Related to #580!

**Comment by ANtlord** [comment]
Hello. Excuse me for dump question. But should I find issues from here? https://github.com/davidhalter/jedi/blob/master/jedi/api/__init__.py#L53

**Comment by davidhalter** [comment]
@ANtlord I don't get the question. Can you rephrase?

**Comment by ANtlord** [comment]
Sorry,  I mean, where should I start to find problem place. I see in example, that completion objects returns by instance of `Script` and after mini research of project, I've solved that I should start search place for adding feature in this class. Am I wrong?

Sorry if my English is not clear.

**Comment by davidhalter** [comment]
Yes, `Script` is a good point to start. Semantic completions (like this one) unfortunately don't have a good layer of abstraction, yet.

**Comment by davidhalter** [comment]
Duplicate of #458
